astropy/config/tests/test_configs.py

```diff
@@ -353,7 +353,7 @@ class Conf(ConfigNamespace):
     with open(f, "wb") as fd:
         apycfg.write(fd)
     with open(f, encoding="utf-8") as fd:
-        lns = [x.strip() for x in fd.readlines()]
+        lns = [x.strip() for x in fd]
```

Comment:
Is `strip()` only used for getting rid of the line endings? There's a few more analogous changes, but I'm only highlighting it here.

Reply:
I don't know, but I wouldn't want this PR to mutate into an unwarranted refactor.

Reply:
On the contrary, now is a very good time for some unwarranted refactoring. Ruff is highlighting code for us that we should look at and might want to improve. If all we ever do is make minimal changes so that Ruff stops complaining then we might still have code that should be improved, but we can't use Ruff to find it anymore.

If addressing the relevant Ruff rule seems to be too complicated or too time consuming then it would be better to add a temporary ignore to `.ruff.toml` and to properly address the rule at some later point.

Reply:
What is your suggestion for substituting `strip()` then? Or does the `in fd` already iterate without the endings?

Reply:
I imagine what @eerovaher has in mind would be something like:
```python
lns = fd.read().splitlines()
```
which does the same thing but is clearer. I'll give it a spin.

Reply:
What I actually had in mind was
```patch
-    with open(f, encoding="utf-8") as fd:
-        lns = [x.strip() for x in fd.readlines()]
-
-    assert "tstnmo = op2" in lns
+    assert "tstnmo = op2" in f.read_text().splitlines()
```
but `f` looks small enough that maybe we could check all of it.

Reply:
Assert statements shouldn't be mixed with business logic as they can (in principle) be "optimized" away with `python -O`

Reply:
But this is a test.

Reply:
oh my, I get so lost in the maze of the discussion

Reply:
done !

Reply:
This refactor was moved to #16656. Please consider closing this thread.